uint8 io
bool active
---
bool result